### PR TITLE
CLI: Update CliClient Node dependency to 10+

### PR DESCRIPTION
**Pull request by joeltaylor**
The CliClient does not build correctly with Node 8 anymore and seems to to be the root of installation failures for many reported issues:
- https://github.com/laurent22/joplin/issues/1107
- https://github.com/laurent22/joplin/issues/2112
- https://github.com/laurent22/joplin/issues/1153

In addition, Node 8 is reaching EOL 2019-12-31

Building and installing with Node 10 succeeds and should alleviate the issues.

### CODE REVIEW

File: CliClient/package-lock.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "joplin",
-  "version": "1.0.149",
+  "version": "1.0.150",
```
Comment (joeltaylor) [inline CliClient/package-lock.json]: I'm not really a Node expert, is this version bump expected when upgrading the node version in `package.json`? If not, I can remove this change.

Comment (tessus) [reply CliClient/package-lock.json]: Yep, this (the version bump) is done during the release process. Please revert this change.

**Comment by joeltaylor** [comment]
@tessus thanks for the feedback -- I'll remember both points in the future! I've reverted the package-lock.json and the html changes.

File: readme/terminal.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ The notes can be [synchronised](#synchronisation) with various targets including
...
+Linux or Windows (via [WSL](https://msdn.microsoft.com/en-us/commandline/wsl/faq?f=255&MSPPError=-2147217396)) | **Important:** First, [install Node 10+](https://nodejs.org/en/download/package-manager/). Node 8 is LTS but not yet available everywhere so you might need to manually install it.<br/><br/>`NPM_CONFIG_PREFIX=~/.joplin-bin npm install -g joplin`<br/>`sudo ln -s ~/.joplin-bin/bin/joplin /usr/bin/joplin`<br><br>By default, the application binary will be installed under `~/.joplin-bin`. You may change this directory if needed. Alternatively, if your npm permissions are setup as described [here](https://docs.npmjs.com/getting-started/fixing-npm-permissions#option-2-change-npms-default-directory-to-another-directory) (Option 2) then simply running `npm -g install joplin` would work.
```
Comment (tessus) [inline readme/terminal.md]: I think we can remove the following sentence: `Node 8 is LTS but not yet available everywhere so you might need to manually install it.`.

@laurent22 what do you think? other than that the PR looks good and should be ready to be merged.

Comment (joeltaylor) [reply readme/terminal.md]: Whoops, I thought I had removed that. Should we remove it entirely or change it to `Node 10 is LTS but not yet available everywhere so you might need to manually install it.`?

I'll take your recommendation here as I'm out of my element with the Node norms.

Comment (tessus) [reply readme/terminal.md]: IMO we should remove it completely. The latest LTS is 12.13.1, so node 10 is most likely available on all systems.
@laurent22 what do you think?

Comment (joeltaylor) [reply readme/terminal.md]: I've gone ahead and removed the line, but can add it back if deemed necessary.

**Comment by tessus** [comment]
@laurent22 This is ready to be merged. The 2 html files don't matter, in fact in this case we don't have to rebuild the web site right away...

**Comment by laurent22** [comment]
I'd prefer if the commits don't include unnecessary files though. I know it might be confusing that they are in the repo (they are required by GitHub Page) but that's why I've added a big warning on top of the files:

```
!!! WARNING !!!
This file was auto-generated from readme/terminal.md and any manual change
made to it will be overwritten. To make a change to this file please modify
the source Markdown file:
https://github.com/laurent22/joplin/blob/master/readme/terminal.md
```

@joeltaylor any chance you could remove the two HTML files please?

**Comment by joeltaylor** [comment]
@laurent22 Most certainly, I originally removed them but must not have force pushed. Sorry about that!

**Comment by joeltaylor** [comment]
HTML files removed, apologies for missing that big warning comment. 👍

**Comment by laurent22** [comment]
Perfect, thanks @joeltaylor!

**Comment by duarteocarmo** [comment]
Hey guys, installing with `brew install joplin`, brew installs node 10.16.3 but I still get 
Screen Shot 2020-03-02 at 17 37 37

**Pull request merged**